modules/core/src/main/java/org/apache/fluo/core/worker/NotificationProcessor.java

```diff
@@ -184,6 +184,8 @@ public void run() {
         // notification should be processed.
         if (notificationFinder.shouldProcess(notification)) {
           workTask.run();
+        } else {
+          notificationProcessed(notification);
```

Comment:
Not having this else was the cause of the bug I saw.  All other changes are misc improvements that were made while I was trying to track down the bug.